### PR TITLE
Serializeable closure support

### PR DESCRIPTION
**Pull request by rubenvanassche**
Serializeable closures can break backtraces, lets fix this!

### CODE REVIEW

File: src/Backtrace.php [hunk 1]
```diff
@@ -171,15 +172,26 @@ protected function toFrameObjects(array $rawFrames): array
         $reduceArgumentsAction = new ReduceArgumentsAction($this->resolveArgumentReducers());
 
         foreach ($rawFrames as $rawFrame) {
-            $frames[] = new Frame(
+            $snippet = null;
+
+            if (substr($currentFile, 0, strlen(ClosureStream::STREAM_PROTO)) === ClosureStream::STREAM_PROTO) {
```
Comment (freekmurze) [inline src/Backtrace.php:177]: Nice use of that constant 👍

File: src/Frame.php [hunk 2]
```diff
@@ -41,22 +51,24 @@ public function __construct(
         $this->class = $class;
 
         $this->applicationFrame = $isApplicationFrame;
+
+        $this->snippet = $snippet;
```
Comment (freekmurze) [inline src/Frame.php]: Maybe rename this to `textSnippet`? To make it really clear that this isn't a `CodeSnippet` object.

File: tests/BacktraceTest.php [hunk 3]
```diff
@@ -157,7 +159,7 @@ public function it_can_get_the_snippet_properties()
         $snippet = $firstFrame->getSnippetProperties(5);
 
         $this->assertStringContainsString('$firstFrame =', $snippet[2]['text']);
-        $this->assertEquals(154, $snippet[2]['line_number']);
+        $this->assertEquals(__LINE__ - 5, $snippet[2]['line_number']);
```
Comment (freekmurze) [inline tests/BacktraceTest.php:162]: Much better!

**Review by freekmurze — CHANGES_REQUESTED** [review]
Looks good! Only some minor nitpicks 👍

**Comment by vovayatsyuk** [comment]
Doesn't the "laravel/serializable-closure" dependency needs to be in the "require" section? It is in the "require-dev" section now.

**Comment by rubenvanassche** [comment]
Good point, taught it was only used in testing but the constant is a problem here. I'll fix it!

**Comment by freekmurze** [comment]
@vovayatsyuk good point

@rubenvanassche It might be better to add a `class_exists` check where appropriate instead of requiring the package.

**Comment by rubenvanassche** [comment]
@freekmurze Yeah good idea!

**Comment by rubenvanassche** [comment]
Fixed

**Pull request merged**